runtime/src/accounts_db.rs

```diff
@@ -4357,13 +4358,29 @@ impl AccountsDb {
                     let sub_storages = snapshot_storages.get(slot);
                     if let Some(sub_storages) = sub_storages {
                         for storage in sub_storages {
-                            let slot = storage.slot();
                             let accounts = storage.accounts.accounts(0);
                             accounts.into_iter().for_each(|stored_account| {
                                 scan_func(LoadedAccount::Stored(stored_account), &mut retval, slot)
                             });
                         }
                     }
+                    if let Some(cache) = accounts_cache {
```

Comment:
@carllin - this is interesting to you. We are using the read cache to augment the data in stores in order to calculate capitalization on banks where we haven't flushed the cache yet.

Reply:
Ah I see, this is because on startup when we run `if !bank_forks.root_bank().calculate_and_verify_capitalization() {` from `do_process_blockstore_from_root()` the cache may not have been flushed yet. This might be a good comment to add to the `calculate_capitalization()` function for why we pass the `use_cache` flag. If we do as this comment  suggests https://github.com/solana-labs/solana/pull/17443/files#r645215712, then that might also be good documentation for that function

Reply:
I don't understand what you linked to. Do you mean `can_cached_slot_be_unflushed`?

Reply:
arg! yeah sorry the links to the comments failed, good thing your intuition is good 😃. Yeah I meant this comment: https://github.com/solana-labs/solana/pull/17443#discussion_r645212955 regarding `can_cached_slot_be_unflushed`.